File: javascript/transpose/transpose.js
export const transpose = (input) => {
  const max_length = input.reduce((acc, curr) => 
    acc = curr.length > acc ? curr.length : acc  , 0);

  return input.reduce((acc, curr, index) => {
    curr.split('').map((l, i) => {
      while(acc[i].length < index) acc[i] += ' ';
      acc[i] += l;
    }); 
    return acc;
  }, Array(max_length).fill(''));
};

File: python/transpose/transpose.py
def transpose(lines):
    if not lines:
        return ''

    lines = lines.split('\n')
    max_line_length = max([len(line) for line in lines])
    output = [[] for _ in range(max_line_length)]
    cur_len_line = 0
    for index in range(max_line_length - 1, -1, -1):
        output[index] = [' ' for _ in range(len(lines))]
        for lindex, line in enumerate(lines):
            if index < len(line):
                output[index][lindex] = line[index]
        # check the last space
        while len(output[index]) > cur_len_line:
            if output[index][-1] == ' ':
                output[index].pop()
            else:
                cur_len_line = len(output[index])
                break

    return '\n'.join([''.join(line) for line in output])
